versioned_docs/version-2021fa/lecture9.md

```diff
@@ -452,7 +454,32 @@ app.listen(port, () => console.log(`Example app listening on port ${port}!`));
 Notice we are having `app.listen` listen for either `process.env.PORT` or port 8080. Previously, we hardcoded `8080` for this parameter. `process.env.PORT` will be defined by Heroku and we want the app to listen for requests on that port in the deployed site.
 :::
 
-We can load the FIREBASE_PRIVATE_KEY from the environment variable instead of having it in plaintext inside the firebase-adminsdk.json. This allows us to publish this code on a public repository without exposing our private key. To run our app locally we create a .env file inside the frontend folder that contains the actual value for the private key. However, in Heroku there is a section called "Config Vars" in the settings which you can use to set the environment variables for deployment.
+We can load the `FIREBASE_PRIVATE_KEY` from the environment variable instead of having it in plaintext inside the `firebase-adminsdk.json` file. This is the service account JSON we were working with with Firebase! By removing the private key from our file, we can publish this code on a public repository without exposing our private key.
+
+```ts title="firebase-adminsdk.json"
+{
+  "type": "service_account",
+  // ...stuff
+  "private_key": "",
+  // ...more stuff
+}
+```
+
+To run our app **locally** we create a .env file inside the frontend folder that contains the actual value for the private key.
+
+Make sure that the `.env` file is inside the `backend` folder!
+
+```env title=".env"
+FIREBASE_PRIVATE_KEY=-----BEGIN PRIVATE KEY-----\n...\n-----END PRIVATE KEY-----\n
+```
+
+However, our `.env` file does not get included with the repository, since it has sensitive information.
```

Comment:
maybe have an aside / tip on how we would go about doing this by making `.gitignore` files here, unless it was covered before?

Reply:
Good idea, I don't think we've ever covered basic git in this course but a blurb here about `.gitignore` would help make it make sense.